lib/record-serialization.js

```diff
@@ -22,10 +23,12 @@ if (jstpNative) {
     jsrs.stringify = serialize;
   }
 } else {
-  console.warn(
-    'JSTP native addon is not built or is not functional. ' +
-    'Run `npm install` in order to build it, otherwise you will get ' +
-    'poor performance.'
-  );
+  if (logger.isWarn()) {
```

Comment:
It isn't possible to replace the logger before this statement will be executed, so it will always use the default logger.